applications/SwimmingDEMApplication/custom_elements/qsvms_dem_coupled.cpp

```diff
@@ -0,0 +1,859 @@
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Joaquin Gonzalez-Usua
+//
+
+// Project includes
+#include "swimming_DEM_application.h"
+#include "qsvms_dem_coupled.h"
+#include "includes/cfd_variables.h"
+#include "includes/checks.h"
+
+#include "../FluidDynamicsApplication/fluid_dynamics_application_variables.h"
+#include "../FluidDynamicsApplication/fluid_dynamics_application.h"
+#include "../FluidDynamicsApplication/custom_utilities/fluid_element_utilities.h"
+#include "custom_utilities/qsvmsdemcoupled_data.h"
+
+namespace Kratos
+{
+
+//////////////////////////Life cycle
+
+template< class TElementData >
+QSVMSDEMCoupled<TElementData>::QSVMSDEMCoupled(IndexType NewId):
+    QSVMS<TElementData>(NewId)
+{}
+
+template< class TElementData >
+QSVMSDEMCoupled<TElementData>::QSVMSDEMCoupled(IndexType NewId, const NodesArrayType& ThisNodes):
+    QSVMS<TElementData>(NewId,ThisNodes)
+{}
+
+
+template< class TElementData >
+QSVMSDEMCoupled<TElementData>::QSVMSDEMCoupled(IndexType NewId, GeometryType::Pointer pGeometry):
+    QSVMS<TElementData>(NewId,pGeometry)
+{}
+
+
+template< class TElementData >
+QSVMSDEMCoupled<TElementData>::QSVMSDEMCoupled(IndexType NewId, GeometryType::Pointer pGeometry, Properties::Pointer pProperties):
+    QSVMS<TElementData>(NewId,pGeometry,pProperties)
+{}
+
+///////////Destructor
+
+template< class TElementData >
+QSVMSDEMCoupled<TElementData>::~QSVMSDEMCoupled()
+{}
+
+template< class TElementData >
+Element::Pointer QSVMSDEMCoupled<TElementData>::Create(IndexType NewId,NodesArrayType const& ThisNodes,Properties::Pointer pProperties) const
+{
+    return Kratos::make_intrusive<QSVMSDEMCoupled>(NewId, this->GetGeometry().Create(ThisNodes), pProperties);
+}
+
+template< class TElementData >
+Element::Pointer QSVMSDEMCoupled<TElementData>::Create(IndexType NewId,GeometryType::Pointer pGeom,Properties::Pointer pProperties) const
+{
+    return Kratos::make_intrusive<QSVMSDEMCoupled>(NewId, pGeom, pProperties);
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+// Inquiry
+
+template< class TElementData >
+int QSVMSDEMCoupled<TElementData>::Check(const ProcessInfo &rCurrentProcessInfo)
+{
+    int out = QSVMS<TElementData>::Check(rCurrentProcessInfo);
+    KRATOS_ERROR_IF_NOT(out == 0)
+        << "Error in base class Check for Element " << this->Info() << std::endl
+        << "Error code is " << out << std::endl;
+
+    // Extra variables
+    KRATOS_CHECK_VARIABLE_KEY(ACCELERATION);
+    KRATOS_CHECK_VARIABLE_KEY(NODAL_AREA);
+
+    // Output variables (for Calculate() functions)
+    KRATOS_CHECK_VARIABLE_KEY(SUBSCALE_VELOCITY);
+    KRATOS_CHECK_VARIABLE_KEY(SUBSCALE_PRESSURE);
```

Comment:
I think that these checks are no longer required after the last changes in the variable class.